zerver/tests/test_auth_backends.py

```diff
@@ -2512,12 +2528,6 @@ def test_get_external_method_dicts_multiple_saml_idps(self) -> None:
                                      'zproject.backends.GitHubAuthBackend',
                                      'zproject.backends.SAMLAuthBackend')
         ):
-            # Calling get_external_method_dicts without a realm returns all methods configured on the server:
-            external_auth_methods = get_external_method_dicts()
-            self.assert_length(external_auth_methods, 3)  # 2 IdP + a dict for github auth
-            self.assertEqual(set([external_auth_methods[0]['name'], external_auth_methods[1]['name']]),
-                             set(['saml:test_idp', 'saml:test_idp2']))
```

Comment:
If feels like we should still be testing this code path, just checking for a different value, right?

Reply:
That'd be reduntant, because the new test `test_get_external_method_dicts_no_realm` already checks that the returned dict should be empty if querying without a realm